### PR TITLE
Add platform agnostic qsort_r

### PR DESCRIPTION
**Pull request by marcbutler**
Provide a portable interface for using qsort_r

Using macros provided a common interface for making use of qsort with a context pointer passed to the comparator which is available in the standard library of all targeted platforms, with argument ordering variations.

### CODE REVIEW

File: src/include/misc.h [hunk 1]
```diff
@@ -227,6 +227,45 @@
...
+ *  const void *rhs
+ *  void *ctx
+ */
+#if defined(__linux__)
+
+#define __wt_qsort_r(base, nmemb, size, compar, ctx) \
+    if (nmemb > 0)                                   \
+    qsort_r(base, nmemb, size, compar, ctx)
+
+#define QSORT_R_ARGS(lhs, rhs, ctx) const void *lhs, const void *rhs, void *ctx
```
Comment (ddanderson) [inline src/include/misc.h:249]: We usually avoid having names without  leading `__`, `WT_` or both for macros.

**Review by ddanderson — COMMENTED** [review]
@marcbutler I think there may be multiple changes (`WT_STRING_MATCH` fix, other bug fixes) mixed into this pull request.  Also, there's no ticket mentioned, so I don't know the scope of the changes.  I think the qsort approach here looks good to me, but is there a motivation to add this?  I don't see it used anywhere except in a test.  I'll be happy to take another look in the context of a ticket.

**Comment by luke-pearson** [comment]
I would like to hold off on this pull request until the other solution, i.e. pulling our own qsort implementation into the codebase is fully explored. But I agree with the points that @ddanderson, if we do end up committing this we'll do it under its own jira ticket.

**Comment by marcbutler** [comment]
Pulling PR as WT-11060 is headed in a different direction.

**Comment by marcbutler** [comment]
If revived need to see what additional possibly accidental diffs have been brought in especially to `misc.h`.

**Pull request closed without merging**